assets/js/src/core/modules/document/actions/translations/components/new-translation-modal.tsx

```diff
@@ -0,0 +1,203 @@
+    {
+      id: currentDocument?.id ?? 0,
+      language: selectedLanguage
+    },
+    { skip: selectedLanguage === '' || isNil(currentDocument?.id) }
+  )
+
+  // Auto-preselect parent when translation parent data is available and contains fullPath
+  useEffect(() => {
+    if (!isNil(translationParentData?.fullPath) && !isNil(translationParentData?.id)) {
+      form.setFieldValue('parent', {
+        id: translationParentData.id,
+        type: 'document',
+        fullPath: translationParentData.fullPath
+      })
+    }
+  }, [translationParentData, form])
+
+  const handleSubmit = async (): Promise<void> => {
+    setIsSubmitting(true)
+    try {
+      const values = await form.validateFields()
+      await onSubmit(values)
+    } finally {
+      setIsSubmitting(false)
+    }
+  }
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const value = e.target.value
+    form.setFieldsValue({
+      title: value,
+      navigation: value,
+      key: value
+    })
+  }
+
+  const handleLanguageChange = (language: string): void => {
+    form.setFieldValue('language', language)
+    setSelectedLanguage(language)
+    // Reset parent field when language changes - it will be auto-filled by useEffect if translation parent exists
+    form.setFieldValue('parent', null)
+  }
+
+  const modalTitle = useInheritance
+    ? t('document.translation.new-document-with-inheritance.modal-title')
+    : t('document.translation.new-document-blank.modal-title')
+
+  return (
+    <WindowModal
+      footer={
+        <ModalFooter>
+          <Button
+            onClick={ onClose }
+            type="default"
+          >
+            {t('cancel')}
+          </Button>
+          <Button
+            loading={ isSubmitting }
+            onClick={ handleSubmit }
+            type="primary"
+          >
+            {t('create')}
+          </Button>
+        </ModalFooter>
+      }
+      onCancel={ onClose }
+      open={ isOpen }
+      size="L"
+      title={ modalTitle }
+    >
+      <Form
+        form={ form }
+        initialValues={ {
+          language: '',
+          parent: null,
+          title: '',
+          navigation: '',
+          key: ''
+        } }
+        layout="vertical"
+      >
+        <Form.Item
+          label={ t('language') }
+          name="language"
+          rules={ [{ required: true, message: t('form.validation.required') }] }
+        >
+          <Select
+            onChange={ handleLanguageChange }
+            options={ availableLanguages }
+          />
+        </Form.Item>
+
+        <Form.Item
+          label={ t('parent') }
+          name="parent"
+          rules={ [{ required: true, message: t('form.validation.required') }] }
+        >
+          <ManyToOneRelation
+            allowToClearRelation
+            documentsAllowed
+            onChange={ (value) => { form.setFieldValue('parent', value) } }
```

Comment:
`onChange` props like this should not be needed as the form values get changed automatically anyways. why do we need this?